tests/TestCase/Http/Client/RequestTest.php

```diff
@@ -37,10 +48,44 @@ public function testConstructorStringData()
         $request = new Request('http://example.com', 'POST', $headers, json_encode($data));
 
         $this->assertEquals('http://example.com', $request->url());
-        $this->assertEquals('POST', $request->getMethod());
+        $this->assertContains($request->getMethod(), 'POST');
+        $this->assertEquals('application/json', $request->getHeaderLine('Content-Type'));
+        $this->assertEquals(json_encode($data), $request->body());
+    }
+    /**
+     * @param array $headers The HTTP headers to set.
+     * @param array|string|null $data The request body to use.
+     * @param string $method The HTTP method to use.
+     *
+     * @dataProvider additionProvider
+     */
+    public function testAdd(array $headers, $data, $method)
+    {
+        $request = new Request('http://example.com', $method, $headers, json_encode($data));
+
+        $this->assertContains($request->getMethod(), $this->methods);
```

Comment:
Wouldn't it be more accurate to check that the method matches exactly? Right now the test would continue to pass even if the request method was transposed to another valid method that wasn't requested.